version: "3.8"

services:

  zookeeper:
    image: confluentinc/cp-zookeeper:latest
    container_name: zookeeper
    environment:
      ZOOKEEPER_CLIENT_PORT: 2181
      ZOOKEEPER_TICK_TIME: 2000

  kafka:
    image: confluentinc/cp-kafka:5.5.1
    container_name: kafka
    depends_on:
      - "zookeeper"
    ports:
      - 9092:9092
    environment:
      KAFKA_BROKER_ID: 1
      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://kafka:29092,PLAINTEXT_HOST://localhost:9092
      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT
      KAFKA_INTER_BROKER_LISTENER_NAME: PLAINTEXT
      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
    volumes:
      - D:\django projects\single_page_app_kafka\data:/tmp
#      - $PWD/data:/tmp/commands_data.txt
#      - $PWD/data:/tmp/feedback_data.txt

  django-kafka-reader:
    image: "python:3.6"
    depends_on:
      - "kafka"
    ports:
      - 8000:8000
    deploy:
      restart_policy:
        condition: "on-failure"
    volumes:
      - $PWD/:/app/ # Update me as you see fit!

    command:
      /bin/bash -c "sleep 10000" #TODO: Update me!

    environment:
      LOG_LEVEL: "info"
      kafka: |
        host: "kafka"
        port: "9092"
        retry_connection: "5"

